pkg/compose/create.go

```diff
@@ -97,11 +91,20 @@ func (s *composeService) create(ctx context.Context, project *types.Project, opt
 	if err := s.ensureNetworks(ctx, project.Networks); err != nil {
 		return err
 	}
-
-	if err := s.ensureProjectVolumes(ctx, project); err != nil {
+	// create observed state for volumes
+	observedVolumesState, err := s.getVolumes(ctx, project.Name)
+	if err != nil {
+		return err
+	}
+	if err := s.ensureProjectVolumes(ctx, project, observedVolumesState, options.RecreateVolumes); err != nil {
 		return err
 	}
 
+	var observedState Containers
```

Comment:
If the optional flag is enabled, we remove the container, so the observedState needs to be moved after `ensureProjectVolumes`